mkosi/resources/mkosi-obs/mkosi.build

```diff
@@ -0,0 +1,107 @@
+
+set -e
+set -x
+
+if [ ! -f /usr/src/packages/SOURCES/mkosi.cpio.rsasign.sig ]; then
+    exit 0
+fi
+
+cpio -t < /usr/src/packages/SOURCES/mkosi.cpio.rsasign.sig
+mkdir -p hashes
+pushd hashes
+cpio -idm < /usr/src/packages/SOURCES/mkosi.cpio.rsasign.sig
+popd
+
+OUTPUTDIR=/work/src/usr/src/packages/OTHER
+
+# OBS signs a hash without certificate information so it cannot simply be
+# attached to the PE binaries, certificate metadata has to be provided separately
+# so we need to create a certutil db and import the certificate manually
+rm -rf nss-db
+mkdir nss-db
+nss_db="$PWD/nss-db"
+certutil -N -d sql:"$nss_db" --empty-password
+certutil -A -d sql:"$nss_db" -n cert -t CT,CT,CT -i /usr/src/packages/SOURCES/_projectcert.crt
+
+cp -r /usr/src/packages/SOURCES/"$IMAGE_ID"* "$OUTPUTDIR"
+rm -f "$OUTPUTDIR/mkosi.cpio.rsasign*" "$OUTPUTDIR"/*.sha*
+
+while read -r SIG
+do
+    export infile="${SIG%.sig}"
+    cpio -i --to-stdout "${infile}" < /usr/src/packages/SOURCES/*.cpio.rsasign > "$(basename "${infile}").sattrs"
+    test -s "$(basename "${infile}").sattrs"
+
+    DEST="$OUTPUTDIR/${SIG#*/}"
+    DEST="${DEST%%.sig}"
+
+    # ensure the EFI hash matches before and after attaching the signature
+    old_hash=$(pesign -n sql:"$nss_db" -h -P -i "/usr/src/packages/SOURCES/$infile" | cut -d ' ' -f1)
+
+    pesign -n sql:"$nss_db" --force -c cert -i "/usr/src/packages/SOURCES/$infile" -o "$DEST" -d sha256 -I "$(basename "${infile}").sattrs" -R "$SIG"
+
+    new_hash=$(pesign -n sql:"$nss_db" -h -i "$DEST" | cut -d ' ' -f1)
+    if [ "$old_hash" != "$new_hash" ]
+    then
+        echo "Pesign hash mismatch error: $old_hash $new_hash"
+        exit 1
+    fi
+
+    rm -f "$(basename "${infile}").sattrs" "$SIG" "$infile"
+done < <(find hashes/ukis -type f \( -name '*efi.sig' -o -name 'vmlinu*.sig' \) -printf '%P\n')
+rm -rf nss-db "$OUTPUTDIR"/*.sig hashes/ukis
+
+while read -r SIG
+do
+    uki="$OUTPUTDIR/$(basename $(dirname ${SIG%.sig}))"
+    pcrs="${uki%.efi}.pcrs"
+    pol="$(basename ${SIG%.sig})"
+
+    test -f "${pcrs}"
+
+    jq -c --arg pol "$pol" --arg sig "$(base64 -w0 < "$SIG")" '
+        to_entries | map(
+            .value |= map(
+                if .pol == $pol then
+                    .sig = $sig
+                else
+                    .
+                end
+            )
+        ) | from_entries
+    ' "${pcrs}" > "${pcrs}.new"
+    mv "${pcrs}.new" "${pcrs}"
+
+    rm -f "$SIG"
+done < <(find hashes/pcrs -type f -name '*.sig')
+rm -rf hashes/pcrs
+
+mkdir -p "$nss_db"
+certutil -N -d sql:"$nss_db" --empty-password
+
+while read -r PCRS
+do
+    uki="${PCRS%.pcrs}.efi"
+    # Attach the content of the PCRS file to the uki with objcopy in a .pcrsig section
+    objcopy --add-section .pcrsig="$PCRS" "$uki" "$uki"
+    mkdir -p hashes/ukis
+    pesign --force -n sql:"$nss_db" -i "$uki" -E "hashes/ukis/$(basename "$uki")"
+done < <(find "$OUTPUTDIR" -type f -name '*.pcrs')
+rm -f "$OUTPUTDIR"/*.pcrs**
```

Comment:
Why the double star here? Does it do anything different without `shopt -s globstar`?